From ca91135428826e389d44eb8d267be1458eb8645b Mon Sep 17 00:00:00 2001
From: Richard Leitner <richard.leitner@skidata.com>
Date: Tue, 13 Nov 2018 12:59:35 +0100
Subject: [PATCH] filter: poppler error: make the msg be const

Poppler v0.70 introduced a const msg for its error functions in
commit 297e6944 ("Error: Make the msg be const"). Therefore adapt
this in the filters.

Signed-off-by: Richard Leitner <richard.leitner@skidata.com>
---
 filter/pdftoijs.cxx            | 4 ++++
 filter/pdftoopvp/pdftoopvp.cxx | 4 ++++
 filter/pdftoraster.cxx         | 4 ++++
 3 files changed, 12 insertions(+)

diff --git a/filter/pdftoijs.cxx b/filter/pdftoijs.cxx
index dd6b6fa0..e69bd326 100644
--- a/filter/pdftoijs.cxx
+++ b/filter/pdftoijs.cxx
@@ -72,7 +72,11 @@ namespace {
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 19
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 23
 void CDECL myErrorFun(void *data, ErrorCategory category,
+#if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 70
+    Goffset pos, const char *msg)
+#else
     Goffset pos, char *msg)
+#endif /* MAJOR > 0 || MINOR >= 70 */
 #else
 void CDECL myErrorFun(void *data, ErrorCategory category,
     int pos, char *msg)
diff --git a/filter/pdftoopvp/pdftoopvp.cxx b/filter/pdftoopvp/pdftoopvp.cxx
index bf25983b..f0d69fae 100644
--- a/filter/pdftoopvp/pdftoopvp.cxx
+++ b/filter/pdftoopvp/pdftoopvp.cxx
@@ -114,7 +114,11 @@ static int outOnePage(PDFDoc *doc, OPVPOutputDev *opvpOut, int pg)
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 19
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 23
 void CDECL myErrorFun(void *data, ErrorCategory category,
+#if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 70
+    Goffset pos, const char *msg)
+#else
     Goffset pos, char *msg)
+#endif /* MAJOR > 0 || MINOR >= 70 */
 #else
 void CDECL myErrorFun(void *data, ErrorCategory category,
     int pos, char *msg)
diff --git a/filter/pdftoraster.cxx b/filter/pdftoraster.cxx
index 49c266ab..975cefdf 100644
--- a/filter/pdftoraster.cxx
+++ b/filter/pdftoraster.cxx
@@ -283,7 +283,11 @@ cmsHPROFILE sgray_profile()
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 19
 #if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 23
 void CDECL myErrorFun(void *data, ErrorCategory category,
+#if POPPLER_VERSION_MAJOR > 0 || POPPLER_VERSION_MINOR >= 70
+    Goffset pos, const char *msg)
+#else
     Goffset pos, char *msg)
+#endif /* MAJOR > 0 || MINOR >= 70 */
 #else
 void CDECL myErrorFun(void *data, ErrorCategory category,
     int pos, char *msg)
